client/test/utils.js

```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  promise: (data) => Promise.resolve().then(() => (data)),
```

Comment:
Nitpick: I think this could be replaced with `promise: Promise.resolve`. (Or we could use `Promise.resolve` in the tests instead of `utils.promise` to avoid indirection.)